editorial: fix broken links to .md files

joshuagl opened the pull request:
Fixes: #938

Signed-off-by: joshua.lock@uk.verizon.com

MarkLodato reviewed (changes requested):
None of these links were broken, except for the one in example.md.

However, there are still many remaining broken links:

https://deploy-preview-946--slsa.netlify.app/provenance/v1.md
Linked from: https://deploy-preview-946--slsa.netlify.app/how-to-orgs

https://deploy-preview-946--slsa.netlify.app/attestation_summary/
Linked from: https://deploy-preview-946--slsa.netlify.app/how-to-orgs

https://deploy-preview-946--slsa.netlify.app/notes/positioning
Linked from: https://deploy-preview-946--slsa.netlify.app/community

https://deploy-preview-946--slsa.netlify.app/_posts/2022-08-29-slsa-github-workflows-generic-ga
Linked from: https://deploy-preview-946--slsa.netlify.app/blog/2023/06/slsa-github-worfklows-container-based

https://docs.google.com/document/d/15Xp8-0Ff_BPg_LMKr1RIKtwAavXGdrgb1BoX4Cl2bE4/edit
Linked from: https://deploy-preview-946--slsa.netlify.app/blog/2023/02/slsa-github-workflows-container-ga and 1 more

https://deploy-preview-946--slsa.netlify.app/blog/2022/09/nist.gov
Linked from: https://deploy-preview-946--slsa.netlify.app/blog/2022/09/eo-in-plain-english

https://deploy-preview-946--slsa.netlify.app/spec/v1.0-rc1/verification_summary
Linked from: https://deploy-preview-946--slsa.netlify.app/spec/v1.0-rc1/ and 10+ more

https://github.com/in-toto/attestation/blob/main/spec/v1.0/README.md
Linked from: https://deploy-preview-946--slsa.netlify.app/spec/v1.0-rc2/provenance and 6 more

https://github.com/in-toto/attestation/blob/main/spec/v1.0/statement.md
Linked from: https://deploy-preview-946--slsa.netlify.app/spec/v1.0-rc2/provenance and 6 more

https://github.com/in-toto/attestation/blob/main/spec/v1.0/field_types.md
Linked from: https://deploy-preview-946--slsa.netlify.app/spec/v1.0-rc2/provenance and 6 more

https://github.com/in-toto/attestation/blob/main/spec/v1.0/resource_descriptor.md
Linked from: https://deploy-preview-946--slsa.netlify.app/spec/v1.0-rc2/provenance and 3 more

https://github.com/in-toto/attestation/blob/main/spec/v1.0/digest_set.md
Linked from: https://deploy-preview-946--slsa.netlify.app/spec/v1.0-rc1/provenance and 2 more

https://deploy-preview-946--slsa.netlify.app/spec/images/supply-chain-threats-build-verification.svg
Linked from: https://deploy-preview-946--slsa.netlify.app/spec/v1.0-rc1/provenance and 2 more

MarkLodato commented:
> None of these links were broken, except for the one in example.md.

Sorry, I was wrong. threats.md was also broken, but only the second and third link. I have no idea why the first one worked but the other two didn't.

joshuagl commented:
I've updated this PR to be a very surgical fix of only the instances of .md extensions in relative links in the generated site.

joshuagl commented:
I added a couple of extra commits and updated the PR description to resolve all of the broken links in https://github.com/slsa-framework/slsa/pull/946#pullrequestreview-1582534781. Or at least, I tried to. @MarkLodato what tool did you use to generate that list of broken links?

MarkLodato reviewed (changes requested):
You can use drlinkcheck.com to check for broken links. Here are the ones I found, besides those that have more specific comments (and besides example.md which is removed in another PR):

1. how-to-orgs.md should be `verification_summary`:
  https://deploy-preview-946--slsa.netlify.app/attestation_summary/
  Linked from: https://deploy-preview-946--slsa.netlify.app/how-to-orgs

2. v0.1/requirements.md: wrap `jdoe@example.com` in backticks to avoid it creating a link
  mailto:jdoe@example.com
  Linked from: https://deploy-preview-946--slsa.netlify.app/spec/v0.1/requirements

3. /blog page should have a trailing slash so that relative links work correctly:
  https://deploy-preview-946--slsa.netlify.app/2023-02-24-slsa-v1-rc
  Linked from: https://deploy-preview-946--slsa.netlify.app/blog

  Should probably send a separate PR to fix this one, since it'll be a code change.

joshuagl commented on docs/_posts/2023-04-03-the-breadth-and-depth-of-slsa.md:
```diff
@@ -4,7 +4,7 @@ author: "Mike Lieberman"
 ---
 
-Interested in getting involved? Now’s the chance to [provide your feedback on the foundational v1 release of the SLSA framework.](2023-02-24-slsa-v1-rc.md)
+Interested in getting involved? Now’s the chance to [provide your feedback on the foundational v1 release of the SLSA framework.](2023-02-24-slsa-v1-rc)
```
Thanks. I wonder if it would be worth switching to the liquid template syntax to automatically generate post links from the filename, i.e. `{% post 2023-02-24-slsa-v1-rc %}`?

I shall update the PR to just use full links from the base url.

MarkLodato reviewed (approved):
Thanks!
